client/src/services/user-settiings.service.ts

```diff
@@ -0,0 +1,23 @@
+import axiosInstance from "./axios.service";
+
+export interface UserDetails {
+    username:string;
+    gender: string;
+    age: string;
+}
+
+export const updateUserDetails = (data: Partial<UserDetails>) => {
+    return axiosInstance.post('/user/', data)
+        .then(response => response.data)
+        .catch(error => {
+            console.error('Error:', error);
+        });
+}
+
+export const getUserDetails = () => {
```

Comment:
Add type here and in `updateUserDetails`

Reply:
done